File: mcp-server/packages/context-extractor-csharp/tests/Examples/UserRepository.cs
namespace ContextExtractor.Tests.Examples;

/// <summary>
/// Repository for user data access.
/// </summary>
public class UserRepository
{
    public DatabaseConnection Db { get; }
    public Logger Logger { get; }

    public UserRepository(DatabaseConnection db, Logger logger)
    {
        Db = db;
        Logger = logger;
    }

    public Dictionary<string, object> GetUserById(int userId)
    {
        return new Dictionary<string, object> { ["id"] = userId };
    }

    public bool SaveUser(Dictionary<string, object> userData)
    {
        return true;
    }

    public bool DeleteUser(int userId)
    {
        return true;
    }

    public List<Dictionary<string, object>> FindUsersByName(string name, int limit = 10)
    {
        return new List<Dictionary<string, object>>();
    }
}

File: mcp-server/packages/context-extractor-java/src/main/java/io/contextextractor/examples/UserRepository.java
package io.contextextractor.examples;

import java.util.List;
import java.util.Map;
import java.util.ArrayList;

/**
 * Repository for user data access.
 */
public class UserRepository {
    private final DatabaseConnection db;
    private final Logger logger;
    
    public UserRepository(DatabaseConnection db, Logger logger) {
        this.db = db;
        this.logger = logger;
    }
    
    public Map<String, Object> getUserById(int userId) {
        return Map.of("id", userId);
    }
    
    public boolean saveUser(Map<String, Object> userData) {
        return true;
    }
    
    public boolean deleteUser(int userId) {
        return true;
    }
    
    public List<Map<String, Object>> findUsersByName(String name, int limit) {
        return new ArrayList<>();
    }
    
    // Getters for testing dependency exploration
    public DatabaseConnection getDb() {
        return db;
    }
    
    public Logger getLogger() {
        return logger;
    }
}
